### ISSUE TITLE
Use Material 3 components everywhere

### ISSUE THREAD

**Issue by papjul** [breezy-weather/breezy-weather#48]
### Describe your suggested feature

Use Material 3 components everywhere

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to **[latest version](https://github.com/breezy-weather/breezy-weather/releases/latest)**.

**Comment by min7-i** [comment]
As briefly mentioned on Matrix the other day, I'm listing a few elements here which aren't fully consistent with material 3 yet. All of them are rather hidden, so it's probably nothing users will notice immediately.

### Time picker
The time picker used to set the time for forecast notifications currently uses a different style instead of the [material time picker](https://m3.material.io/components/time-pickers/overview).

<details><summary>screenshots</summary>

| current version | material version |
| --- | --- |
| ![time picker](https://github.com/breezy-weather/breezy-weather/assets/97251923/3c02afa3-0689-4a87-9ad8-e8af5c2199c8) | ![grafik](https://github.com/breezy-weather/breezy-weather/assets/97251923/2e21d41c-80eb-43ed-9486-579431b6b45c) |

</details>

As far as I could see, the current time picker is also not following the time format (12h vs 24h format) selected in Android settings. It's always using the 24h format.

_According to the material 3 docs, the jetpack compose version of the time picker is currently only available in the experimental version and it seems that some elements are still missing there. However, a release version does exist for the Material Components for Android library._

### Snackbars and floating action buttons
Snackbars are currently displayed in front of FABs instead of displaying them above the FABs. ([m3 reference](https://m3.material.io/components/snackbar/guidelines#f762d356-69a1-402c-ab69-3621c23f91d2))

However, this is only affecting the location list when the FAB to add the current location is visible and one of the locations is marked as resident city.

<details><summary>screenshots</summary>

| current behavior | material 3 behavior |
| --- | --- |
| ![Image_1](https://github.com/breezy-weather/breezy-weather/assets/97251923/1b075a55-9eb3-4347-89c7-20ff94e9d12d) | ![grafik](https://github.com/breezy-weather/breezy-weather/assets/97251923/921b00e9-8eed-492e-8189-785383f646bb) |

</details>

**Comment by papjul** [comment]
Do we have other Material 3 components to rewrite or can we just keep #128 for the snackbar?

**Comment by min7-i** [comment]
I found a few things related to Material 3/compose most of which are probably better addressed in separate issues, but I wanted to list them here first so we can decide how to proceed with them:

### color of navigation bar
When using the 3-button navigation bar it has a solid color in the location list, location search and a shade in settings. That's probably due to the fact that the app currently doesn't have proper edge-to-edge support. As this will be enforced with [Android 15](https://developer.android.com/about/versions/15/behavior-changes-15#window-insets) it's probably a good idea to check this as part of #837 which would also allow us to get rid of some deprecated features in `DisplayExtensions` such as [systemUiVisibility](https://github.com/breezy-weather/breezy-weather/blob/2aa8a07c8b28b6ee02f28eaf12051319adb026d2/app/src/main/java/org/breezyweather/common/extensions/DisplayExtensions.kt#L139).

### compose
The following layouts are not yet implemented in compose, but I don't know if #334 is the correct issue for those as that is more about inconsistencies with day/night mode:

1. settings to reorder header details, daily/hourly trends and cards
2. location list items in management fragment
3. icon pack viewer ([PreviewIconActivity](https://github.com/breezy-weather/breezy-weather/blob/main/app/src/main/java/org/breezyweather/settings/activities/PreviewIconActivity.kt))

No. 1 could serve as a pilot for implementing the required swipe and drag&drop gestures in compose which could then be reused in the location list (2).

**Comment by at-pyrix** [comment]
Also, an option to just have a solid background instead of the default one would be great! 


Something like this: ([Bura](https://f-droid.org/packages/com.davidtakac.bura/))

![Uploading Screenshot_20240918-234232_Droid-ify.jpg…]()

**Comment by papjul** [comment]
@min7-i do we still have things to do here after #1336?

From what I understand, only a rewrite in Compose remains, which is not required IMO, as it is supported and works fine as is. Btw, there would be a lot of work to rewrite the entire main screen.

Last time I tried to implement swipe and drag&drop gestures in Compose on the location list, it made the code more complex and I wasn't able to complete it. But trying first on settings reorder is a good idea if you want to take a look at it?

**Comment by min7-i** [comment]
> do we still have things to do here after #1336?

> Snackbars are currently displayed in front of FABs instead of displaying them above the FABs.

This is still happening because in the compose layouts the snackbar is not properly integrated into the layout, but simply attached to the [root view](https://github.com/breezy-weather/breezy-weather/blob/08d373cf5009118927351c38465cabef5d291f83/app/src/main/java/org/breezyweather/common/snackbar/Snackbar.kt#L574-L578). However, I don't think there is an easy fix for this. In my opinion it's fine to accept the current behavior as it's only a minor issue.

Apart from that, I think we've covered everything.

> But trying first on settings reorder is a good idea if you want to take a look at it?

I want to finish some other things first, but if nobody else is working on it by then, I'll look into it.
